be/src/olap/olap_server.cpp

```diff
@@ -314,6 +314,7 @@ void StorageEngine::_compaction_tasks_producer_callback() {
     ProfilerRegisterThread();
 #endif
     LOG(INFO) << "try to start compaction producer process!";
+    int64_t interval = config::generate_compaction_tasks_min_interval_ms;
```

Comment:
You defined it as a mutable config, you should reassign `interval ` in while loop.

Reply:
It sounds reasonable.